pkg/controller/atlasdatabaseuser/databaseuser_test.go

```diff
@@ -52,46 +52,46 @@ func TestCheckUserExpired(t *testing.T) {
 	fakeClient := fake.NewClientBuilder().WithScheme(scheme).Build()
 
 	t.Run("Validate DeleteAfterDate", func(t *testing.T) {
-		result := checkUserExpired(zap.S(), fakeClient, "", *mdbv1.DefaultDBUser("ns", "theuser", "").WithDeleteAfterDate("foo"))
+		result := checkUserExpired(context.TODO(), zap.S(), fakeClient, "", *mdbv1.DefaultDBUser("ns", "theuser", "").WithDeleteAfterDate("foo"))
 		assert.False(t, result.IsOk())
 		assert.Equal(t, reconcile.Result{}, result.ReconcileResult())
 
-		result = checkUserExpired(zap.S(), fakeClient, "", *mdbv1.DefaultDBUser("ns", "theuser", "").WithDeleteAfterDate("2021/11/30T15:04:05"))
+		result = checkUserExpired(context.TODO(), zap.S(), fakeClient, "", *mdbv1.DefaultDBUser("ns", "theuser", "").WithDeleteAfterDate("2021/11/30T15:04:05"))
```

Comment:
see comment above, and all TODO()s in all tests.

Reply:
Will fix, but I will not go further. We might want to prepare a follow up PR for ensuring all tests use Background consistently everywhere, because we have as many "TODOs" in tests as "Backgrounds".

Reply:
sounds good 👍 Feel free to submit a ticket or augment any existing one.